database/database.go

```diff
@@ -153,6 +154,50 @@ func (db DatastoreBacked) StateForDomain(domain string) (state DomainState, err
 	return state, nil
 }
 
+// StatesForDomains returns the domains states for the given domains
+func (db DatastoreBacked) StatesForDomains(domains []string) (states []DomainState, err error) {
+	// Set up datastore context
+	c, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client, datastoreErr := db.backend.NewClient(c, db.projectID)
+	if datastoreErr != nil {
+		return nil, datastoreErr
+	}
+
+	getDomainStates := func(keys []*datastore.Key) ([]DomainState, error) {
+		domainStates := make([]DomainState, len(keys))
+		if dsErr := client.GetMulti(c, keys, domainStates); dsErr != nil {
+			return nil, dsErr
+		}
+		for i := range domainStates {
+			domainStates[i].Name = keys[i].Name
+		}
+		return domainStates, nil
+	}
+
+	var keys []*datastore.Key
+	var res []DomainState
+	for _, domain := range domains {
+		key := datastore.NameKey(domainStateKind, domain, nil)
+		keys = append(keys, key)
+		if len(keys) >= batchSize {
+			if tempStates, err := getDomainStates(keys); err != nil {
+				return nil, err
+			} else {
```

Comment:
This is because tempstates is declared and assigned in the if statement, you can declare tempStates one line earlier if you want to remove the else statement, e.g.:
```
if len(keys) >= batchSize {
    var tempStates []DomainState
    if tempStates, err := getDomainStates(keys); err != nil {
```